src/shared/lib/LazyMobXCache.ts

```diff
@@ -135,17 +147,30 @@ export default class LazyMobXCache<Data, Query> {
         this.updateCache(toMerge);
     }
 
-    private putData(queries:Query[], data:Data[]) {
-        const toMerge:Cache<Data> = {};
+    private putData(queries:Query[], data:FetchResult<Data, Metadata>) {
+        const toMerge:Cache<Data, Metadata> = {};
         const keyHasData:{[key:string]:boolean} = {};
 
-        for (const datum of data) {
-            const datumKey = this.dataToKey(datum);
-            toMerge[datumKey] = {
-                status: "complete",
-                data: datum
-            };
-            keyHasData[datumKey] = true;
+        for (const dataElt of data) {
```

Comment:
right, the example case is if you want to augment data with the study that it came from, if that study id is not echoed in the API data. its cheaper to wrap a list of data that all have the same metadata (study id), rather than wrap each individual datum. so all the data in that AugmentedData gets associated with that metadata via it being added to the CacheData and it being passed into dataToKey to find the key